### PR TITLE
chore(deps): align deps, remove items from depalign ignores

### PR DESCRIPTION
**Pull request by Anemy**
Reduces the dependencies we have listed in `.depalignrc.json`.
Aligns some dependencies we had not aligned, moving them to the highest existing version in our packages.

Some help from 
```bash
npm run depalign
```
which runs https://github.com/mongodb-js/devtools-shared/blob/6c524413ccc9f8f2ac5d47820b2dd8dc0fecf337/packages/monorepo-tools/src/depalign.ts

### CODE REVIEW

File: packages/compass-app-stores/package.json [hunk 1]
```diff
@@ -57,8 +57,8 @@
     "@mongodb-js/mocha-config-compass": "^1.3.9",
     "@mongodb-js/prettier-config-compass": "^1.0.2",
     "@mongodb-js/tsconfig-compass": "^1.0.4",
-    "@testing-library/dom": "8.20.1",
-    "@testing-library/react": "12.1.5",
+    "@testing-library/dom": "^8.20.1",
+    "@testing-library/react": "^12.1.5",
```
Comment (Anemy) [inline packages/compass-app-stores/package.json:61]: Drive-by, noticed these dependencies were exact matches while we usually use the `Compatible with version caret` https://docs.npmjs.com/cli/v8/configuring-npm/package-json#dependencies

File: package-lock.json [hunk 2]
```diff
@@ -45008,6 +45000,23 @@
         "node": ">=8"
       }
     },
+    "packages/compass-e2e-tests/node_modules/debug": {
+      "version": "4.3.4",
```
Comment (gribnoysup) [inline package-lock.json:45004]: Hmm, weird that this is getting added here if we aligned all the versions 🤔

**Review by gribnoysup — COMMENTED** [review]
Seems like debug ended up being installed in nested workspace node_modules instead of being hoisted. Might make sense to manually hoisting it by running `npm install debug@<version>` at the root of repository and then removing it from the package.json (similar to how we make sure that electron is hoisted when updated)

File: package-lock.json (hunk 2 above)
Comment (Anemy) [reply package-lock.json]: Nice catch, ran the manual hosting steps we do in `update-electron`. Looks like it got it this time.

**Review by gribnoysup — APPROVED** [review]
Nice clean-up!
